src/test/spec/retryable_writes/mod.rs

```diff
@@ -0,0 +1,374 @@
+                            &test_case.description
+                        ));
+
+                        let description = &test_case.description;
+                        if let Some(contain) = expected_labels.error_labels_contain {
+                            contain.iter().for_each(|label| {
+                                assert!(
+                                    error.contains_label(label),
+                                    "{}: error labels should include {}",
+                                    description,
+                                    label,
+                                );
+                            });
+                        }
+
+                        if let Some(omit) = expected_labels.error_labels_omit {
+                            omit.iter().for_each(|label| {
+                                assert!(
+                                    !error.contains_label(label),
+                                    "{}: error labels should not include {}",
+                                    description,
+                                    label,
+                                );
+                            });
+                        }
+                    }
+                };
+            }
+
+            let coll_name = match test_case.outcome.collection.name {
+                Some(name) => name,
+                None => coll_name.to_string(),
+            };
+
+            let coll = client.get_coll_with_options(&db_name, &coll_name, options);
+            let options = FindOptions::builder().sort(doc! { "_id": 1 }).build();
+            let actual_data: Vec<Document> = coll
+                .find(None, options)
+                .await
+                .unwrap()
+                .try_collect()
+                .await
+                .unwrap();
+            assert_eq!(test_case.outcome.collection.data, actual_data);
+
+            if let Some(fail_point) = test_case.fail_point {
+                let disable = doc! {
+                    "configureFailPoint": fail_point.get_str("configureFailPoint").unwrap(),
+                    "mode": "off",
+                };
+                client
+                    .database("admin")
+                    .run_command(disable, None)
+                    .await
+                    .unwrap();
+            }
+        }
+    }
+
+    let _guard = LOCK.run_exclusively().await;
+    run_spec_test(&["retryable-writes"], run_test).await;
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn transaction_ids_excluded() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+
+    if !(client.is_replica_set() || client.is_sharded()) {
+        return;
+    }
+
+    let coll = client.init_db_and_coll(function_name!(), "coll").await;
+
+    let excludes = |command_name: &str| -> bool {
```

Comment:
can we name this to `excludesTxnNumber`? Also, why is the txnNumber excluded from these operations?